### PR TITLE
fix broken precision value test

### PR DESCRIPTION
**Pull request by clD11**
This PR fixes the broken precision value test

### CODE REVIEW

File: plugins/ccxtExchange_test.go [hunk 1]
```diff
@@ -556,24 +557,17 @@ func TestCancelOrder_Ccxt(t *testing.T) {
 }
 
 func TestGetOrderConstraints_Ccxt_Precision(t *testing.T) {
-	// coinbasepro gives incorrect precision values so we do not test it here
```
Comment (nikhilsaraf) [inline plugins/ccxtExchange_test.go:559]: Can we leave this comment in here

File: plugins/ccxtExchange_test.go [hunk 2]
```diff
@@ -556,24 +557,17 @@ func TestCancelOrder_Ccxt(t *testing.T) {
...
 	testCases := []struct {
 		exchangeName       string
 		pair               *model.TradingPair
 		wantPricePrecision int8
 		wantVolPrecision   int8
 	}{
 		{
-			// disable ccxt-kraken based tests for now because of the 403 Forbidden Security check API error
```
Comment (nikhilsaraf) [inline plugins/ccxtExchange_test.go:567]: Let’s leave this comment in here too. It’s a TODO for me but I forgot to mark as such.
Can prefix with // TODO NS

File: plugins/ccxtExchange_test.go [hunk 3]
```diff
@@ -587,28 +581,20 @@ func TestGetOrderConstraints_Ccxt_Precision(t *testing.T) {
...
-				[]api.ExchangeAPIKey{emptyAPIKey},
-				[]api.ExchangeParam{emptyParams},
-				[]api.ExchangeHeader{},
-				false,
-				getEsParamFactory(kase.exchangeName),
-			)
-			if !assert.NoError(t, e) {
-				return
-			}
+	for _, tc := range testCases {
```
Comment (nikhilsaraf) [inline plugins/ccxtExchange_test.go:584]: Can we leave the var as kase to keep consistent with rest of codebase?
Tc is probably a better abbreviation/var-name but can migrate to that later in one sweeping change

File: plugins/ccxtExchange_test.go [hunk 4]
```diff
@@ -587,28 +581,20 @@ func TestGetOrderConstraints_Ccxt_Precision(t *testing.T) {
...
-			}
-			if !assert.Equal(t, kase.wantVolPrecision, result.VolumePrecision) {
-				return
-			}
+			epf := getEsParamFactory(tc.exchangeName)
+
+			ccxtExchange, e := makeCcxtExchange(tc.exchangeName, nil, []api.ExchangeAPIKey{emptyAPIKey},
+				[]api.ExchangeParam{emptyParams}, []api.ExchangeHeader{}, false, epf)
+
+			require.NoError(t, e)
```
Comment (nikhilsaraf) [inline plugins/ccxtExchange_test.go:592]: Good idea to use require, it’s exactly the behavior we want. We should migrate to using require in other places too long-term. I’ll do that at some point

File: plugins/ccxtExchange_test.go [hunk 5]
```diff
@@ -556,24 +557,17 @@ func TestCancelOrder_Ccxt(t *testing.T) {
...
-			// 	pair:               &model.TradingPair{Base: model.XLM, Quote: model.USD},
-			// 	wantPricePrecision: 6,
-			// 	wantVolPrecision:   8,
-			// }, {
 			exchangeName:       "binance",
 			pair:               &model.TradingPair{Base: model.XLM, Quote: model.USDT},
-			wantPricePrecision: 5,
-			wantVolPrecision:   1,
+			wantPricePrecision: 4,
+			wantVolPrecision:   4,
```
Comment (nikhilsaraf) [inline plugins/ccxtExchange_test.go:570]: Can you double check volume precision, I think it should be 0, not 4

**Review by nikhilsaraf — COMMENTED** [review]
Comments inline

File: plugins/ccxtExchange_test.go (hunk 5 above)
Comment (clD11) [reply plugins/ccxtExchange_test.go:577]: there is some logic that sets it to the price when its 0 
	volumePrecision := ccxtMarket.Precision.Amount
	if volumePrecision == 0 {
		volumePrecision = ccxtMarket.Precision.Price
	}
	
	so it ends up at 4

**Review by nikhilsaraf — APPROVED** [review]
Lgtm, thanks

**Comment by nikhilsaraf** [comment]
Failing because of glide so bypassing CI to merge. We are fixing glide issue by migrating to go modules in a separate PR

**Pull request merged**